frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts

```diff
@@ -0,0 +1,55 @@
+import * as path from 'path';
+import { browser, ExpectedConditions as until } from 'protractor';
+import { click } from '@console/shared/src/test-utils/utils';
+import { clickNavLink } from '@console/internal-integration-tests/views/sidenav.view';
+import {
+  createItemButton,
+  isLoaded,
+  saveChangesBtn,
+} from '@console/internal-integration-tests/views/crud.view';
+import { VM_CREATE_AND_EDIT_TIMEOUT_SECS } from './utils/constants/common';
+import * as cdiUploadView from '../views/cdiUploadView';
+import { inputPVCName, inputPVCSize } from '@console/internal-integration-tests/views/storage.view';
+import {
+  selectItemFromDropdown,
+  storageclassDropdown,
+} from '@console/ceph-storage-plugin/integration-tests/views/pvc.view';
+
+describe('KubeVirt CDI Upload', () => {
+  const pvc = { name: 'upload-pvc', size: 1, unit: 'Gi', storageClassName: 'rook-ceph' };
+
+  it(
+    'ID() Upload data to CDI',
+    async () => {
+      const fileToUpload = '../views/cdiUploadView.ts';
+      const absolutePath = path.resolve(__dirname, fileToUpload);
+
+      await clickNavLink(['Storage', 'Persistent Volume Claims']);
+      await isLoaded();
+      await click(createItemButton);
+      await click(cdiUploadView.uploadCdiFormButton);
+
+      await browser.wait(
+        until.textToBePresentInElement(
+          $('.co-m-pane__heading'),
+          'Upload Data to Persistent Volume Claim',
+        ),
+      );
+
+      await inputPVCName.sendKeys(pvc.name);
+      await inputPVCSize.sendKeys(pvc.size);
+      await selectItemFromDropdown(pvc.storageClassName, storageclassDropdown);
+
+      // firefox needs the input to be shown
+      await browser.executeAsyncScript(function(callback) {
+        // You can use any other selector
+        (document.querySelector('input[type="file"]') as HTMLElement).style.display = 'inline';
+        callback();
+      });
+      await cdiUploadView.uploadInput.sendKeys(absolutePath);
+      await click(saveChangesBtn);
+      await browser.wait(until.textToBePresentInElement(cdiUploadView.uploadProgress, '100%'));
```

Comment:
can we start a VM/container and check the file/content is there?

Reply:
this is outside our scope, the responsibility of this UI section is to upload a file, what you're talking about is the CDI responsibility.
We can however go to the PVCs page and check that the pvc is there and bounded

Reply:
start a VM needs the upload file is a truly vm image, we can do it in tier2 test for this card: https://issues.redhat.com/browse/CNV-6020.

Reply:
it is not out of the scope, because we are doing final integration tests here. And we can't just depend on a simple status to check that everything went well.

ok we can move it to tier2 tests then @gouyang can you create a description for this test please?

Reply:
add a comment to https://issues.redhat.com/browse/CNV-6020

Reply:
added comment @gouyang